### DISCUSSION TITLE
failed to load kernel modules after a+b update

### DISCUSSION THREAD

**Discussion by ghost** [rauc/rauc#1317] [Q&A]
**Describe the bug**
Performed an a+b rootfs update and rauc successfully installed the bundle.
After rebooting the board the the bootloader booted from the new slot.
But when the kernel booted; it could not find the kernel modules.
Inspecting the kernel revision id; we see the correct kernel revision id with the commit id
Inspecting the /lib/modules directory; we see the correct revision id but the commit is different.

From the above; why did rauc not update the commit if of the kernel?
Also is rauc performed a comlete update or partial?

**Comment by ejoerns** [comment]
@pritpalm-kuk RAUC replaces full (file system) images. If I understood it correctly, kernel and modules are all part of the rootfs?
Thus if there is an inconsistency, this must be produced by your Embedded Linux build system, not by RAUC.

**Comment by ghost** [comment]
ejoerns the kernel and modules are a part of the rootfs.
what we dont seem to understand is when we use the command uname -r we get the kernel revision; but the commit id is different to the directory in /lib/modules

**Comment by ejoerns** [comment]
@pritpalm-kuk you should be able to check if this is already the case for the image image contained in your bundle.
If you do not have it anymore, use `unsquashfs` or `rauc extract` to extract the bundle content.
RAUC does not modify the image contents.

Since I have no insight into your rootfs creation, it is quite hard for me to help you to debug this.

**Reply by ghost** [reply]
rootfs_A is in mmcblk2p2
rootfs_B is in mmcblk2p3

The above partitions do not contain the kernel and dtb's. The kernel and dtb's are defined in the following partition:
boot in mmcblk2p1

From the I also want rauc to update mmcblk2p1; hence kernel updates.
How do I configure rauc to achieve.
Our wic file is defined below:
part u-boot --source rawcopy --sourceparams="file=imx-boot" --ondisk mmcblk --no-table --align 32
part /boot --source bootimg-partition --ondisk mmcblk --fstype=vfat --label boot --active --align 8192 --size 64
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_A --fixed-size=2G --align 8192
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_B --fixed-size=2G --align 8192
part /data --source rootfs --rootfs-dir=${IMAGE_ROOTFS}/data --ondisk mmcblk --fstype=ext4 --size 100K --label data --align 8192
part /caam --source rootfs --rootfs-dir=${IMAGE_ROOTFS}/caam --ondisk mmcblk --fstype=ext4 --size 10K --label fresco --align 8192

bootloader --ptable msdos

From the above if I create a secondary  /boot partition; how can it be made active after we perform an OTA?
